### PR TITLE
Add basic drawer component

### LINKED ISSUE #62: Drawer
Element documentation: https://element.eleme.io/#/en-US/component/drawer

Components to build:
- Basic usage
- Open with nested form

Props:
- Props for setting width 
- prop to indicate the direction of the sliding transition

Use:
Drawer component will be used for Adding Workspace and seeing Workspace list and settings

### PR DESCRIPTION
**Pull request by ninetteadhikari**
## What is the Purpose?
- Basic drawer component with props for:
  - Panel header
  - Drawer direction
  - Drawer size
  - Show close button

## What was the approach?
Used Element UI components

## Are there any concerns to addressed further before or after merging this PR?
None

## Mentions?
@andrewtpham 

## Issue(s) affected?
Resolves #62

### CODE REVIEW

**Comment by michaelbukachi** [comment]
Hey @ninetteadhikari nice work!.
There's only one issue I'm facing at the moment. Once I open the drawer and close. I can't seem to be able to open it again. 
Also, would be possible to add a prop enable/disable clicking outside the drawer hide to it?

**Comment by ninetteadhikari** [comment]
> Hey @ninetteadhikari nice work!.
> There's only one issue I'm facing at the moment. Once I open the drawer and close. I can't seem to be able to open it again.
> Also, would be possible to add a prop enable/disable clicking outside the drawer hide to it?

thanks @michaelbukachi 😊I added some styling and added props to enable/disable closing the modal by clicking on the shadow and also show/not show the close button. However I wasn't able to resolve why the button wasn't working after the first click :( may be it has something to do with the syncing `visible.sync` Would be great if you can take a look. Thank you!!! 🙏

**Comment by michaelbukachi** [comment]
@ninetteadhikari Might it because props don't react the same way as data values. Maybe consider using watcher instead to watch the prop.

**Comment by ninetteadhikari** [comment]
> @ninetteadhikari Might it because props don't react the same way as data values. Maybe consider using watcher instead to watch the prop.

ah I haven't used watcher before! I tried to add a watcher when the Drawer visibility changes but it keeps giving this error that I can't mutate a prop. What do you think? 🤔 
image

**Comment by michaelbukachi** [comment]
Ah. What you need to do is have an inner data variable maybe `_drawerVisible`. In the watcher, instead of changing the value of `drawerVisible` change the value of the inner variable. Then change the value `visible.sync` to the inner variable.

File: src/stories/Drawer.stories.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    }
+  },
+  template: `
+  <div>
+    <el-button @click="drawer = true" type="primary" style="margin-left: 16px;">
+      Open drawer
+    </el-button>
+    <basic-drawer
+      :title="title"
+      :drawerVisible="$data._drawer"
```
Comment (ninetteadhikari) [inline src/stories/Drawer.stories.js:30]: @michaelbukachi is this how we add inner variable? sorry for all the back and forth :( I am very new to all this.

Comment (michaelbukachi) [reply src/stories/Drawer.stories.js:30]: Nope. It should be:
```
:drawerVisible="_drawer"
```
then somewhere in the script section under the data field:
```
data: function () {
  return {
    _drawer: false
  }
}
```

File: src/stories/Drawer.stories.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      title: 'I am the title'
+    }
+  },
+  watch: {
+    drawer (newValue, oldValue) {
+      if (!newValue) {
+        this.drawer = false
+      }
+    }
+  },
```
Comment (michaelbukachi) [inline src/stories/Drawer.stories.js:22]: This code is supposed to be in the `BasicDrawer` component not outside.:sweat_smile:

**Pull request merged**